File: backend/src/main/java/com/evalorithm/entity/ExamResult.java
package com.evalorithm.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "exam_results",
       uniqueConstraints = @UniqueConstraint(columnNames = {"exam_id", "student_profile_id"}))
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ExamResult extends BaseEntity {

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "exam_id", nullable = false)
    private Exam exam;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "student_profile_id", nullable = false)
    private StudentProfile studentProfile;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "attempt_id")
    private ExamAttempt attempt;

    private Double totalMarksObtained;

    private Integer totalMarksPossible;

    private Double percentage;

    private String grade;

    @Column(nullable = false)
    private Boolean isPassed;

    private Integer correctAnswers;

    private Integer wrongAnswers;

    private Integer skippedQuestions;

    private Integer timeTakenMinutes;

    private LocalDateTime evaluatedAt;
}

File: android/app/src/main/java/com/evalorithm/data/model/ExamResult.kt
package com.evalorithm.data.model

data class ExamResult(
    val id: Long,
    val examTitle: String,
    val studentName: String,
    val totalMarksObtained: Double,
    val totalMarksPossible: Int,
    val percentage: Double,
    val grade: String,
    val isPassed: Boolean,
    val correctAnswers: Int,
    val wrongAnswers: Int,
    val skippedQuestions: Int,
    val timeTakenMinutes: Int,
    val evaluatedAt: String?
)
